src/errors.rs

```diff
@@ -8,6 +8,9 @@ use std::sync::Arc;
 /// Error types within Rahmen
 #[derive(std::fmt::Debug)]
 pub enum RahmenError {
+    /// unknown case for conversion
+    /// TODO it's NYI
+    CaseUnknown,
```

Comment:
Change to `CaseUnknown(String)`, and include the error message you're generating in the error variant.